### ISSUE TITLE
Only noise when training from scratch on COCO dataset

### ISSUE THREAD

**Issue by alphacoder01** [CompVis/stable-diffusion#735]
Hi, I'm trying to train a stable-diffusion from scratch on COCO dataset. Till now it's completed 190k steps but still the output of the model is complete noise. Does anyone have any idea regarding how much more should I train to see some outputs.

**Here's the train.yaml file I'm using:**

```
model:
  base_learning_rate: 1.0e-04
  target: ldm.models.diffusion.ddpm.LatentDiffusion
  params:
    linear_start: 0.00085
    linear_end: 0.0120
    num_timesteps_cond: 1
    log_every_t: 200
    timesteps: 1000
    first_stage_key: "jpg"
    cond_stage_key: "caption"
    image_size: 32
    channels: 4
    cond_stage_trainable: false   # Note: different from the one we trained before
    conditioning_key: crossattn
    monitor: val/loss_simple_ema
    scale_factor: 0.18215
    use_ema: true

    scheduler_config: # 10000 warmup steps
      target: ldm.lr_scheduler.LambdaLinearScheduler
      params:
        warm_up_steps: [ 0 ]
        cycle_lengths: [ 10000000000000 ] # incredibly large number to prevent corner cases
        f_start: [ 1.e-6 ]
        f_max: [ 1. ]
        f_min: [ 1. ]

    unet_config:
      target: ldm.modules.diffusionmodules.openaimodel.UNetModel
      params:
        image_size: 32 # unused
        in_channels: 4
        out_channels: 4
        model_channels: 320
        attention_resolutions: [ 4, 2, 1 ]
        num_res_blocks: 2
        channel_mult: [ 1, 2, 4, 4 ]
        num_heads: 8
        use_spatial_transformer: True
        transformer_depth: 1
        context_dim: 768
        use_checkpoint: True
        legacy: False

    first_stage_config:
      target: ldm.models.autoencoder.AutoencoderKL
      params:
        embed_dim: 4
        monitor: val/rec_loss
        ddconfig:
          double_z: true
          z_channels: 4
          resolution: 256
          in_channels: 3
          out_ch: 3
          ch: 128
          ch_mult:
          - 1
          - 2
          - 4
          - 4
          num_res_blocks: 2
          attn_resolutions: []
          dropout: 0.0
        lossconfig:
          target: torch.nn.Identity

    cond_stage_config:
      target: ldm.modules.encoders.modules.FrozenCLIPEmbedder

data:
  target: main.DataModuleFromConfig
  params:
    batch_size: 32
    num_workers: 16
    train:
      target: dataset.COCO
      params:
        path: MYPATH
        split: train
        min_resize_res: 256
        max_resize_res: 256
        crop_res: 256
        flip_prob: 0.5
    validation:
      target: dataset.COCO
      params:
        path: MYPATH
        split: val
        min_resize_res: 256
        max_resize_res: 256
        crop_res: 256

lightning:
  callbacks:
    image_logger:
      target: main.ImageLogger
      params:
        batch_frequency: 2000
        max_images: 2
        increase_log_steps: False

  trainer:
    max_epochs: 50000
    benchmark: True
    accumulate_grad_batches: 4
    check_val_every_n_epoch: 4
``` 

I'm using the default inference.yaml file that is provided in the repository

**Training Command**
- CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python main.py --name coco_ldm --base configs/stable-diffusion/train.yaml --train --gpus 0,1,2,3,4,5,6,7 --logdir ~/scratch/Experiments/COCO/ --scale_lr False 

`I had to specify the scale_lr = False in the training command as without it, the loss went to NaN within few steps.`

Here is the output after 190k steps of training (using the txt2img.py script provided with default params)
![image](https://github.com/CompVis/stable-diffusion/assets/33857516/c4a0dab5-cca7-4309-9f8d-2e1b44f0637b)

**Comment by jenuk** [comment]
Doesn't look like you are providing a first stage checkpoint. The first stage encodes an image into a smaller space, in which the diffusion model is then trained. So in order, for that training to work, you need to provide a meaningful space. The first stage is frozen during second-stage training.  So either start of by training a first stage from scratch or use one of the first stages provided in the latent diffusion repository.

**Comment by Q1ngS0ng** [comment]
> Hi, I'm trying to train a stable-diffusion from scratch on COCO dataset. Till now it's completed 190k steps but still the output of the model is complete noise. Does anyone have any idea regarding how much more should I train to see some outputs.
> 
> **Here's the train.yaml file I'm using:**
> 
> ```
> model:
>   base_learning_rate: 1.0e-04
>   target: ldm.models.diffusion.ddpm.LatentDiffusion
>   params:
>     linear_start: 0.00085
>     linear_end: 0.0120
>     num_timesteps_cond: 1
>     log_every_t: 200
>     timesteps: 1000
>     first_stage_key: "jpg"
>     cond_stage_key: "caption"
>     image_size: 32
>     channels: 4
>     cond_stage_trainable: false   # Note: different from the one we trained before
>     conditioning_key: crossattn
>     monitor: val/loss_simple_ema
>     scale_factor: 0.18215
>     use_ema: true
> 
>     scheduler_config: # 10000 warmup steps
>       target: ldm.lr_scheduler.LambdaLinearScheduler
>       params:
>         warm_up_steps: [ 0 ]
>         cycle_lengths: [ 10000000000000 ] # incredibly large number to prevent corner cases
>         f_start: [ 1.e-6 ]
>         f_max: [ 1. ]
>         f_min: [ 1. ]
> 
>     unet_config:
>       target: ldm.modules.diffusionmodules.openaimodel.UNetModel
>       params:
>         image_size: 32 # unused
>         in_channels: 4
>         out_channels: 4
>         model_channels: 320
>         attention_resolutions: [ 4, 2, 1 ]
>         num_res_blocks: 2
>         channel_mult: [ 1, 2, 4, 4 ]
>         num_heads: 8
>         use_spatial_transformer: True
>         transformer_depth: 1
>         context_dim: 768
>         use_checkpoint: True
>         legacy: False
> 
>     first_stage_config:
>       target: ldm.models.autoencoder.AutoencoderKL
>       params:
>         embed_dim: 4
>         monitor: val/rec_loss
>         ddconfig:
>           double_z: true
>           z_channels: 4
>           resolution: 256
>           in_channels: 3
>           out_ch: 3
>           ch: 128
>           ch_mult:
>           - 1
>           - 2
>           - 4
>           - 4
>           num_res_blocks: 2
>           attn_resolutions: []
>           dropout: 0.0
>         lossconfig:
>           target: torch.nn.Identity
> 
>     cond_stage_config:
>       target: ldm.modules.encoders.modules.FrozenCLIPEmbedder
> 
> data:
>   target: main.DataModuleFromConfig
>   params:
>     batch_size: 32
>     num_workers: 16
>     train:
>       target: dataset.COCO
>       params:
>         path: MYPATH
>         split: train
>         min_resize_res: 256
>         max_resize_res: 256
>         crop_res: 256
>         flip_prob: 0.5
>     validation:
>       target: dataset.COCO
>       params:
>         path: MYPATH
>         split: val
>         min_resize_res: 256
>         max_resize_res: 256
>         crop_res: 256
> 
> lightning:
>   callbacks:
>     image_logger:
>       target: main.ImageLogger
>       params:
>         batch_frequency: 2000
>         max_images: 2
>         increase_log_steps: False
> 
>   trainer:
>     max_epochs: 50000
>     benchmark: True
>     accumulate_grad_batches: 4
>     check_val_every_n_epoch: 4
> ```
> 
> I'm using the default inference.yaml file that is provided in the repository
> 
> **Training Command**
> 
> * CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python main.py --name coco_ldm --base configs/stable-diffusion/train.yaml --train --gpus 0,1,2,3,4,5,6,7 --logdir ~/scratch/Experiments/COCO/ --scale_lr False
> 
> `I had to specify the scale_lr = False in the training command as without it, the loss went to NaN within few steps.`
> 
> Here is the output after 190k steps of training (using the txt2img.py script provided with default params) ![image](https://user-images.githubusercontent.com/33857516/241189198-c4a0dab5-cca7-4309-9f8d-2e1b44f0637b.png)

Could you tell me about your COCO dataset format? Do you convert the original COCO dataset to a CSV format with two columns which is ‘jpg’ and ‘caption’?

**Comment by alphacoder01** [comment]
@Q1ngS0ng That's correct!

**Comment by alphacoder01** [comment]
> Doesn't look like you are providing a first stage checkpoint. The first stage encodes an image into a smaller space, in which the diffusion model is then trained. So in order, for that training to work, you need to provide a meaningful space. The first stage is frozen during second-stage training. So either start of by training a first stage from scratch or use one of the first stages provided in the latent diffusion repository.

@jenuk Thanks for your help, this was actually the issue. Correcting this fixed everything.

**Comment by TorAP** [comment]
@alphacoder01 can you shed some light on how to train it from scratch? Do you just supply a config.yaml? or how is the process?

**Comment by ppbrown** [comment]
@alphacoder01 how about sharing your input dataset, and the model, on huggingface, even if it isnt impressive?
I would love to learn from your efforts, rather than having to needlessly re-do what has already been done by others
